[pytorch][refactor] Explicitly use auto* for pointers

jianyuh opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#32548 [pytorch][refactor] Explicitly use auto* for pointers**

As Title says.

Differential Revision: [D19541893](https://our.internmc.facebook.com/intern/diff/D19541893/)

jianyuh committed:
[pytorch][refactor] Explicitly use auto* for pointers

As Title says.

Differential Revision: [D19541893](https://our.internmc.facebook.com/intern/diff/D19541893/)

[ghstack-poisoned]

kostmo commented:
## :pill: CircleCI build failures summary and remediations
As of commit 56852a3f:


* **1/1** failures introduced in this PR


## Detailed failure analysis

One may explore the probable reasons each build failed interactively [on the Dr. CI website](https://dr.pytorch.org/commit-details.html?sha1=56852a3f2eee7f7881e524e3319877915e021fcd).

### :detective: 1 new failure recognized by patterns
The following build failures do not appear to be due to upstream breakage:
#### [![See CircleCI build](https://avatars0.githubusercontent.com/ml/7?s=12)](https://circleci.com/gh/pytorch/pytorch/4326591) pytorch_xla_linux_xenial_py3_6_clang7_build (1/1)
**Step:** "Build" ([full log](https://dr.pytorch.org/api/view-log-full?build_id=89987254) | [pattern match details](https://dr.pytorch.org/build-details.html?build_id=89987254))
<details>
<summary>
<code>Jan 23 20:07:01 RuntimeError: Failed to apply patch: /var/lib/jenkins/workspace/xla/torch_patches/X10-device_test.diff</code>
</summary>

```
Jan 23 20:07:01 File to patch:  
Jan 23 20:07:01 Skip this patch? [y]  
Jan 23 20:07:01 4 out of 4 hunks ignored 
Jan 23 20:07:01 Applying patch file: /var/lib/jenkins/workspace/xla/torch_patches/X10-clip_grad.diff 
Jan 23 20:07:01 Applying patch file: /var/lib/jenkins/workspace/xla/torch_patches/X10-device_test.diff 
Jan 23 20:07:01 Traceback (most recent call last): 
Jan 23 20:07:01   File "/var/lib/jenkins/workspace/xla/scripts/cond_patch.py", line 67, in <module> 
Jan 23 20:07:01     patch_repo(args) 
Jan 23 20:07:01   File "/var/lib/jenkins/workspace/xla/scripts/cond_patch.py", line 49, in patch_repo 
Jan 23 20:07:01     raise RuntimeError('Failed to apply patch: {}'.format(ppath)) 
Jan 23 20:07:01 RuntimeError: Failed to apply patch: /var/lib/jenkins/workspace/xla/torch_patches/X10-device_test.diff 
Jan 23 20:07:01 + cleanup 
Jan 23 20:07:01 + retcode=1 
Jan 23 20:07:01 + set +x 
Jan 23 20:07:01 =================== sccache compilation log =================== 
Jan 23 20:07:01 =========== If your build fails, please take a look at the log above for possible reasons =========== 
Jan 23 20:07:01 Compile requests                 1 
Jan 23 20:07:01 Compile requests executed        0 
Jan 23 20:07:01 Cache hits                       0 
Jan 23 20:07:01 Cache misses                     0 
Jan 23 20:07:01 Cache timeouts                   0 
```

</details>



---

<details><summary>This comment was automatically generated by <a href="https://github.com/kostmo/circleci-failure-tracker/tree/master/docs/from-pull-request-comment">Dr. CI</a> (expand for details).</summary>Follow <a href="https://dr.pytorch.org/admin/comments-opt-out.html">this link to opt-out</a> of these comments for your Pull Requests.

Please report bugs/suggestions on the <a href="https://github.com/kostmo/circleci-failure-tracker/issues">GitHub issue tracker</a>.
</details>

jianyuh committed:
Update on "[pytorch][refactor] Explicitly use auto* for pointers"


As Title says.

Differential Revision: [D19541893](https://our.internmc.facebook.com/intern/diff/D19541893/)

[ghstack-poisoned]

jianyuh committed:
Update on "[pytorch][refactor] Explicitly use auto* for pointers"


As Title says.

Differential Revision: [D19541893](https://our.internmc.facebook.com/intern/diff/D19541893/)

[ghstack-poisoned]

jianyuh committed:
Update on "[pytorch][refactor] Explicitly use auto* for pointers"


As Title says.

Differential Revision: [D19541893](https://our.internmc.facebook.com/intern/diff/D19541893/)

[ghstack-poisoned]
